cypress/component/column-linking-table.cy.js

```diff
@@ -0,0 +1,228 @@
+        },
+
+        dataDictionary: {
+
+            original: {
+
+                "age": { "Description": "age of the participant" },
+                "sex": { "Description": "sex of the participant as reported by the participant" },
+                "group": { "Description": "diagnostic status determined by the study clinician at baseline" },
+                "group_dx": { "Description": "specific diagnosis determined by the study clinician at baseline" },
+                "number_comorbid_dx": { "Description": "a number of diagnoses comorbid with UD (e.g., GAD, PTSD)" },
+                "medload": { "Description": "reflects the number of dosage of psychotropic medications taken by participants. Higher numbers correspond to more medications and/or higher medication dosage " },
+                "iq": { "Description": "IQ derived based on the NART assessment." },
+                "session": { "Description": "scanning session" }
+            }
+        },
+
+        dataTable: {
+
+            columns: [
+
+                "participant_id",
+                "age",
+                "sex",
+                "group",
+                "group_dx",
+                "number_comorbid_dx",
+                "medload",
+                "iq",
+                "session"
+            ],
+
+            original: [{
+
+                "participant_id": "",
+                "age": "",
+                "sex": "",
+                "group": "",
+                "group_dx": "",
+                "number_comorbid_dx": "",
+                "medload": "",
+                "iq": "",
+                "session": ""
+            }]
+        },
+
+        toolColorPalette: {
+
+            color1: "category-style-0",
+            color2: "category-style-1",
+            color3: "category-style-2",
+            color4: "category-style-3",
+            color5: "category-style-4",
+            colorDefault: "category-style-default"
+        }
+    },
+
+    actions: {
+
+        dispatch: (p_actionName, p_payload) => { }
+    },
+
+    getters: {}
+};
+
+store.getters = {
+
+    categories: () => store.state.categories,
+    categoryClasses: () => store.state.categoryClasses,
+    columnToCategoryMap: () => store.state.columnToCategoryMap,
+    dataDictionary: () => store.state.dataDictionary,
+    dataTable: () => store.state.dataTable
+};
+
+const props = {
+
+    selectedCategory: "Subject ID"
+};
+
+// Tests
+
+describe("Tests basic functionality of the table that links categories with data table columns", () => {
+
+    it("Link one column from the currently selected category", () => {
+
+        // 1. Arrange - Set up the spy, mount the component, and bind the spy to it
+        const onColumnNameClickedSpy = cy.spy().as("onColumnNameClickedSpy");
+        cy.spy(store.actions, "dispatch").as("dispatchSpy");
+        cy.mount(ColumnLinkingTable, {
+
+            mocks: {
+
+                $store: store.actions
+            },
+
+            computed: store.getters,
```

Comment:
I don't think the getters are used by the component at all. Is this an oversight here?

Reply:
The mock store and getters have been completely reworked as per our discussion in Slack huddle.

See https://github.com/neurobagel/annotation_tool/pull/230/commits/01377f20aa87029aef545dad0ddc73b7c9baf76b